packages/docusaurus-theme-classic/src/theme/DocPage/index.tsx

```diff
@@ -93,14 +94,29 @@ function DocPageContent({
             {hiddenSidebar && (
               <div
                 className={styles.collapsedDocSidebar}
-                title="Expand sidebar"
-                aria-label="Expand sidebar"
+                title={translate({
+                  id: 'theme.DocPageContent.expandButtonLabel',
+                  message: 'Expand sidebar',
+                  description:
+                    'The ARIA label and title attribute for expand button of doc sidebar',
+                })}
+                aria-label={translate({
+                  id: 'theme.DocPageContent.expandButtonLabel',
+                  message: 'Expand sidebar',
+                  description:
+                    'The ARIA label and title attribute for expand button of doc sidebar',
+                })}
```

Comment:
maybe factorize?

Reply:
Yes, I agree, fixed.

By the way, I thought, maybe we shouldn't use component names in `id` messages? This is not a very flexible approach if we change the structure of components in the future. Is it better to use short domain-specific names instead like `docs` instead `DocPageContent`?

Reply:
Some examples:

- theme.DocPaginator.navAriaLabel -> theme.docs.paginator.navAriaLabel
- theme.BlogListPaginator.ariaLabel -> theme.blog.post.paginator.ariaLabel
- theme.BlogPostItem.tags -> theme.blog.post.tags
- theme.BlogPostPaginator.ariaLabel -> theme.blog.paginator.ariaLabel

- theme.CodeBlock.copyButtonAriaLabel -> theme.common.copyCodeButtonAriaLabel (as the copy code button can be used not only in docs)

Reply:
Yes, agree with that idea.
Do you want to change the label to more meaningful ids before the i18n release?

> theme.blog.paginator.ariaLabel

I don't like this kind of id too much, as it does not really give an idea of what in the paginator you are translating for example.

Reply:
> Do you want to change the label to more meaningful ids before the i18n release?

Yes, why not, but just need to decide on the naming convention

> I don't like this kind of id too much, as it does not really give an idea of what in the paginator you are translating for example.

Hmm, what are the alternatives then? I actually think this is more understandable -- the `blog` (by default, in list) has a `paginator` (or `pagination`) for which an `aria label` is set that you need to translate.

Reply:
I don't know what a "paginator" is in this context, for me the pagination is next/previous and there are 2 labels to translate, not just 1 😅  I think the id should make it clearer about what label we are talking about, maybe including somehow the original English string if short?

Reply:
It is assumed to refer to the root element. Moreover, we will still have a hint in the form of the `description` key. 

```
  "theme.blog.post.paginator.ariaLabel": {
    "message": "Blog post page navigation",
    "description": "The ARIA label for the blog posts pagination"
  },
```

I would also not include the original content in the id (as much as possible), since it can change over time.

> theme.CodeBlock.copyButtonAriaLabel -> theme.common.copyCodeButtonAriaLabel (as the copy code button can be used not only in docs) OR theme.codeBlock.copyButtonAriaLabel OR theme.common.codeBlock.copyButtonAriaLabel - not sure which is the best of these.

And what do you think about this? Which option do you like better here?

Reply:
> It is assumed to refer to the root element.

I see now. As it's a `nav`, why not `navAriaLabel` instead?

> Moreover, we will still have a hint in the form of the description key.

That's true, but at runtime and in default translation bundles we don't have the description, so having good ids is still convenient for debugging :)

> I would also not include the original content in the id (as much as possible), since it can change over time.

Agree, I meant it for buttons like `nextButtonAriaLabel` etc, it's unlikely the next button  becomes something else in the future ;)

> And what do you think about this? Which option do you like better here?

I like `theme.codeBlock.copyButtonAriaLabel` or `theme.common.copy`

I'd keep the namespace `theme.common` for small simple strings like actions, yes/no etc

Reply:
> I see now. As it's a nav, why not navAriaLabel instead?

Yep, I even mentioned similar one for docs - theme.docs.paginator.navAriaLabel

> Agree, I meant it for buttons like nextButtonAriaLabel etc, it's unlikely the next button becomes something else in the future ;)

But in fact now we sometimes use exactly English content as a key:

![image](https://user-images.githubusercontent.com/4408379/107805020-c8a83480-6d75-11eb-998d-b307c28dccef.png)

If we refactor name `theme.BlogPostPaginator.newerPost`, then which one is better? `theme.blog.post.paginator.nextButton.text` or `theme.blog.post.paginator.nextButtonText` (eg, could be `nextButtonAriaLabel`, `nextButtonTitle` in future).

Reply:
I think `theme.blog.post.paginator.nextButton.text` is fine, particularly if there's any chance that we might need any other translated text on that button (like  an aria label)

Reply:
In the commit https://github.com/facebook/docusaurus/pull/4219/commits/b3f0dfcacb6b1278be2c899c5060d1bd9f31ec71 I changed the keys of some messages.
In the messages of specific components (like PwaReloadPopup), I left the ids as it. But if the translation message is intended for a blog or docs, then the different name convention that we discussed here applies to it. In addition, there is `common` namespace for simple messages.

I also added new messages to French translation, so you can translate them if you are satisfied with the current result.
The `base.json` file is supposed to serve as the basis for new translations (which I hope we will get from the community).

Reply:
Thanks, that looks good.
Not very available today, will review it more tomorrow morning